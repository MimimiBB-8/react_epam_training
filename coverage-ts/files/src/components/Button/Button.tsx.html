
  <!DOCTYPE html>
  <html>
    <head>
      <title>Button.tsx</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="negative"><td class="">src\components\Button\Button.tsx</td><td class="">79.17%</td><td class="">80%</td><td class="">24</td><td class="">19</td><td class="">5</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import style from &#x27;./Button.module.scss&#x27;

interface ButtonProps {
  classname?: string
  title?: string
  onClick?: (param: any) =&gt; void
  type?: any
}

function Button({ classname = &#x27;base_button&#x27;, title, onClick, type}: ButtonProps) {
  return (
    &lt;&gt;
      &lt;button className={`${style[classname]}`} onClick={onClick} type={type}&gt;
        {title}
      &lt;/button&gt;
    &lt;/&gt;
  )
}

export default Button
</textarea><pre id="annotations" style="display:none">[{&quot;file&quot;:&quot;src\\components\\Button\\Button.tsx&quot;,&quot;line&quot;:0,&quot;character&quot;:7,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\Button\\Button.tsx&quot;,&quot;line&quot;:5,&quot;character&quot;:13,&quot;text&quot;:&quot;param&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\Button\\Button.tsx&quot;,&quot;line&quot;:6,&quot;character&quot;:2,&quot;text&quot;:&quot;type&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\Button\\Button.tsx&quot;,&quot;line&quot;:9,&quot;character&quot;:61,&quot;text&quot;:&quot;type&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\Button\\Button.tsx&quot;,&quot;line&quot;:12,&quot;character&quot;:28,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1}]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Sun, 12 Mar 2023 16:12:31 GMT</p>
    </body>
  </html>
  