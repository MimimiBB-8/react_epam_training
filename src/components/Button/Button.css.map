{"version":3,"sourceRoot":"","sources":["Button.sass"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;;;AAEF;EACE;EACA;EACA;;;AAEF;EACE;EACA","file":"Button.css"}