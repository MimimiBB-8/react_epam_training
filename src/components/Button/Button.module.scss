button {
  height: 57px;
  width: 180px;
  border-radius: 4px;
  font-weight: 500;
  font-size: 20px;
  line-height: 24px;
  text-align: center;
  text-transform: uppercase;
  box-sizing: border-box;
  border: 1.5px solid #f65261;
}



.base_button {
  background-color: #f65261;
  right: 0;
}

.reset_button {
  color: #f65261;
  background-color: transparent;
  margin-right: 13px;
}

.search_button {
  width: 233px;
  background-color: #f65261;
}

.close_button {
  position: absolute;
  top: 15px;
  right: 28px;
  width: 22px;
  height: 22px;
  background-color: transparent;
  border: none;
}

.close_button:before,
.close_button:after {
  content: '';
  position: absolute;
  width: 21px;
  height: 1px;
  border: 1px solid #ffffff;
  background: #ffffff;
  border-radius: 20px;
}

.close_button:before {
  -webkit-transform: rotate(45deg);
  transform: rotate(45deg);
}

.close_button:after {
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

