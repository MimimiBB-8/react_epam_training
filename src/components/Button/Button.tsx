import style from './Button.module.scss'

interface ButtonProps {
  classname?: string
  title?: string
  onClick?: (param: any) => void
  type?: any
}

function Button({ classname = 'base_button', title, onClick, type}: ButtonProps) {
  return (
    <>
      <button className={`${style[classname]}`} onClick={onClick} type={type}>
        {title}
      </button>
    </>
  )
}

export default Button
