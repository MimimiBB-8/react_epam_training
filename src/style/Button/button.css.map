{"version":3,"sourceRoot":"","sources":["button.sass"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;;;AAEF;EACE;EACA;EACA","file":"button.css"}