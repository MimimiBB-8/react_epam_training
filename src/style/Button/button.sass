button
  height: 57px
  width: 180px
  border-radius: 4px
  font-weight: 500
  font-size: 20px
  line-height: 24px
  text-align: center
  text-transform: uppercase
  box-sizing: border-box
  border: 1.5px solid #F65261

.base_button
  background-color: #F65261
  right: 0

.reset_button
  color: #F65261
  background-color: transparent
  margin-right: 13px

.search_button
  width: 233px
  background-color: #F65261
