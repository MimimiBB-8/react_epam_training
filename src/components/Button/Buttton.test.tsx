import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Button from './Button';


describe('component button', () => {
    it('render button', () => {
        const onClick = jest.fn()

        const { asFragment } = render(
            <Router>
                <Button onClick={onClick} />
            </Router>);

        expect(asFragment()).toMatchSnapshot()
    })
    it('render button with title', () => {
        const onClick = jest.fn()
         render(
            <Router>
                <Button onClick={onClick} title='search' />
            </Router>);
        expect(screen.getByText('search')).toBeInTheDocument
    })
})