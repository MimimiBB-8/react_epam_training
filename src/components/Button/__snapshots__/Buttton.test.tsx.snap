// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component button render button 1`] = `
<DocumentFragment>
  <button
    class="base_button"
  />
</DocumentFragment>
`;
